src/boot/efi/vmm.c

```diff
@@ -295,9 +289,14 @@ const char* smbios_find_oem_string(const char *name) {
         left -= type11->header.length;
 
         for (const char *p = s; p < s + left; ) {
-                const char *e = memchr(p, 0, s + left - p);
-                if (!e || e == p) /* Double NUL byte means we've reached the end of the OEM strings. */
+                // Found double NUL
+                if (p[0] == '\0' && p[1] == '\0') {
+                        p += 2;
                         break;
+                }
+
+                // Find end of string (should always exist, every string is NUL terminated)
+                const char *e = ASSERT_PTR(memchr(p, 0, s + left - p));
```

Comment:
nope, this should not be an assert. firmware is shit, we must handle this gracefully.